packages/oscal-react-library/src/components/OSCALMetadataCommon.tsx

```diff
@@ -1,12 +1,405 @@
+
+export const OSCALMetadataAddress: React.FC<OSCALMetadataAddressProps> = (props) => {
+  const tryFormatAddress = (address: Address) => {
+    const lines = address["addr-lines"] ?? [];
+    const cityAndState = [address.city, address.state].filter((it) => it).join(", ");
+    const line = [cityAndState, address["postal-code"]].filter((it) => it).join(" ");
+    const allLines = [...lines, line, address.country].filter((it) => it);
+    return allLines;
+  };
+  const { address } = props;
+
+  const addr = tryFormatAddress(address);
+  const formatted = addr.flatMap((it, index) => [it, <br key={index} />]);
+
+  return addr.length ? (
+    <TextWithIcon icon={getAddressIcon(address.type)}>{formatted}</TextWithIcon>
+  ) : (
+    <TextWithIcon icon={<ErrorOutlineIcon />}>
+      <Typography variant="body2">Address cannot be formatted</Typography>
+    </TextWithIcon>
+  );
+};
+
+export interface OSCALMetadataEmailProps {
+  email: string;
+}
+
+export const OSCALMetadataEmail: React.FC<OSCALMetadataEmailProps> = (props) => {
+  return (
+    <Typography>
+      <Link href={`mailto:${props.email}`}>{props.email}</Link>
+    </Typography>
+  );
+};
+
+export interface OSCALMetadataTelephoneProps {
+  telephone: TelephoneNumber;
+}
+
+export const OSCALMetadataTelephone: React.FC<OSCALMetadataTelephoneProps> = (props) => {
+  const { telephone } = props;
+
+  return (
+    <TextWithIcon icon={getPhoneIcon(telephone.type)}>
+      <Link href={`tel:${telephone.number}`}>{telephone.number}</Link>
+    </TextWithIcon>
+  );
+};
+
+export enum MetadataInfoType {
+  address = "address",
+  telephone = "telephone",
+  email = "email",
+}
+
+/**
+ * Maps a contact list item to the correct component.
+ *
+ * @param infoProps Props to pass to the component
+ * @param type Contact info type to use for map
+ */
+const mapContactType = (infoProps: Address | TelephoneNumber | string, type: MetadataInfoType) => {
+  switch (type) {
+    case MetadataInfoType.email:
+      return <OSCALMetadataEmail email={infoProps as string} />;
+    case MetadataInfoType.address:
+      return <OSCALMetadataAddress address={infoProps as Address} />;
+    case MetadataInfoType.telephone:
+      return <OSCALMetadataTelephone telephone={infoProps as TelephoneNumber} />;
+  }
+};
+
+export interface MetadataInfoListProps {
+  /**
+   * List of metadata contact info
+   */
+  list: Address[] | string[] | TelephoneNumber[] | undefined;
+  /**
+   * Type of metadata contact info
+   */
+  infoType: MetadataInfoType;
+  /**
+   * Message to display if list is empty
+   */
+  emptyMessage: string;
+}
+
+export const MetadataInfoList: React.FC<MetadataInfoListProps> = (props) => {
+  const { list, infoType, emptyMessage } = props;
+
+  if (!list?.length) {
+    return <Typography> {emptyMessage} </Typography>;
+  }
+
+  return (
+    <>
+      {list.map((item, index) => {
+        return <ListItem key={index}>{mapContactType(item, infoType)}</ListItem>;
+      })}
+    </>
+  );
+};
```

Comment:
I think I am starting to realize that this is a bit overly generic to expose as an interface in the library. And it requires passing two tightly coupled properties with the same value.


I'd rather see us do:

```tsx
export type Email = string;

interface LocationDisplayProps<T> {
  contact: T;
}

type LocationDisplay<T> = React.FC<LocationDisplayProps<T>>;

const OSCALMetadataAddress: LocationDisplay<Address> = ({ contact }) => {
  // The same code as now, just using `contact` instead of address
};
const OSCALMetadataEmail: LocationDisplay<Email> = ({ contact }) => {
};
const OSCALMetadataTelephoneNumber: LocationDisplay<TelephoneNumber> = ({ contact }) => {
};

interface MetadataLocationListProps<T> {
  list: T[] | undefined; // I do not like the name list.
  component: LocationDispay<T>;
  emptyMessage: string;
}

const MetadataLocationList<T>: React.FC<MetadataInfoListProps<T>> = (props) => {
  const { list, emptyMessage } = props;
  const LocationDisplayComponent = props.component;

  if (!list?.length) {
    return <NotSpecifiedTypography>{emptyMessage}</NotSpecifiedTypography>
  }

  return (
    <>
      {list.map((item, index) => {
        <ListItem key={index}><LocationDisplayComponent contact={item} /></ListItem>
      })}
    </>
  );
};
```